.github/workflows/rust-sdk.yml

```diff
@@ -0,0 +1,84 @@
+name: 'Rust SDK'
+
+on:
+    push:
+        branches: ['main']
+    pull_request:
+        types: [opened, reopened, synchronize, assigned, review_requested]
+        paths:
+            - 'sdk/highlight-rust/**'
+            - '.github/workflows/rust-sdk.yml'
+
+permissions:
+    checks: write
+
+defaults:
+    run:
+        working-directory: ./sdk/highlight-rust
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+jobs:
+    build:
+        name: Build
+        runs-on: ubuntu-latest
+        if: github.event_name == 'pull_request'
+        defaults:
+            run:
+                working-directory: ./sdk/highlight-rust
+        steps:
+            - name: Checkout Code
+              uses: actions/checkout@v3
+              with:
+                  fetch-depth: 0
+
+            - name: Install Rust toolchain
+              uses: actions-rs/toolchain@v1
+              with:
+                  toolchain: stable
+                  override: true
+
+            - name: Build highlightio
+              uses: katyo/publish-crates@v2
+              with:
+                  path: ./sdk/highlight-rust/
+                  dry-run: true
+                  args: -p highlightio
+
+            - name: Build highlightio-actix
+              uses: katyo/publish-crates@v2
+              with:
+                  path: ./sdk/highlight-rust/
+                  dry-run: true
+                  args: -p highlightio-actix
+    publish:
+        name: Publish
+        runs-on: ubuntu-latest
+        if: github.event_name == 'push'
```

Comment:
awesome, so the dry run will happen in branches, but the cargo publish will only actually happen on main push.